doc/source/whatsnew/v0.18.0.txt

```diff
@@ -102,6 +103,37 @@ And multiple aggregations
    r.agg({'A' : ['mean','std'],
           'B' : ['mean','std']})
 
+.. _whatsnew_0180.enhancements.rangeindex:
+
+Range Index
+^^^^^^^^^^^
+
+A ``RangeIndex`` has been added to the ``Int64Index`` sub-classes to support a memory saving alternative for common use cases. This has simliar implementation to the python ``xrange`` object in that it only stores the start, stop, and step values for the index. It will transparently interact with the user API, converting to ``Int64Index`` if needed. (:issue:`939`)
```

Comment:
Maybe also explicitly state that this is now the default index when constructing a Series/DataFrame without specifying the index?